### PR TITLE
Add REQUIRE, trim exports, work on 0.4

### LINKED ISSUE #13: Missing REQUIRE file
(StatsBase, Distributions)

### LINKED ISSUE #14: Stop exporting `rand`

### PR DESCRIPTION
**Pull request by garborg**
Closes #13, closes #14.

### CODE REVIEW

File: src/hyper_log_log.jl [hunk 1]
```diff
@@ -1,4 +1,8 @@
-hash32(d::Any) = uint32(hash(d))
+if VERSION < v"0.4.0-"
+    hash32(d::Any) = uint32(hash(d))
+else
+    hash32(d::Any) = hash(d) % Uint32
+end
```
Comment (StefanKarpinski) [inline src/hyper_log_log.jl:5]: Use `Compat` instead and just define as `hash(d) % UInt32`?

Comment (garborg) [inline src/hyper_log_log.jl:5]: You got my hopes up, but:

``` julia
julia> using Compat

julia> d = 1151551
1151551

julia> hash(d) % UInt32
ERROR: `rem` has no method matching rem(::Uint64, ::Type{Uint32})
```

Comment (StefanKarpinski) [inline src/hyper_log_log.jl:5]: Could be added to `Compat`, no?

**Comment by eytan** [comment]
As a heads up, a number of these changes are already in my PR, https://github.com/johnmyleswhite/StreamStats.jl/commit/eb7b894bd4b9b199be5d940d1652a8e437a3d85c

File: src/hyper_log_log.jl (hunk 1 above)
Comment (garborg) [inline src/hyper_log_log.jl:5]: Is that as simple as

``` julia
Base.rem{T <: Real}(x::Real, t::Type{T}) = convert(t, x) 
```

?

**Comment by garborg** [comment]
Thanks, @eytan. If you're planning to release that soon, certainly feel free to wait and incorporate the rest of the changes there.

If it might be a while, it would be good to tag a patchlevel release to squash the 0.4 bug and make the package work for users who don't already have Distributions installed.

**Comment by johnmyleswhite** [comment]
Should be handled by commit by @eytan.

**Comment by garborg** [comment]
Great!

**Pull request closed without merging**